helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl

```diff
@@ -235,7 +235,37 @@ receivers:
               from: log
               to: $$
   {{- end }}
-
+  # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/journaldreceiver
+  {{- if (eq (include "splunk-otel-collector.logsEnabled" .) "true") }}
+  {{- if and (eq .Values.logsEngine "otel") .Values.logsCollection.journaldLogs.enabled }}
+  {{- if .Values.logsCollection.journaldLogs.units }}
+  {{- range $_, $unit := .Values.logsCollection.journaldLogs.units }}
+  {{- printf "journald/%s:" $unit.name | nindent 2 }}
+    directory: {{ $.Values.logsCollection.journaldLogs.directory }}
+    units: [{{ $unit.name }}]
+    priority: {{ $unit.priority | default $.Values.logsCollection.journaldLogs.defaultPriority }}
+    resource:
+      com.splunk.source: {{ $.Values.logsCollection.journaldLogs.directory }}
+      com.splunk.sourcetype: 'EXPR("kube:"+$$._SYSTEMD_UNIT)'
+      com.splunk.index: {{ $.Values.logsCollection.journaldLogs.index | default $.Values.splunkPlatform.index}}
+      host.name: 'EXPR(env("K8S_NODE_NAME"))'
+      {{- if $.Values.clusterName }}
+      k8s.cluster.name: {{ $.Values.clusterName }}
+      {{- end }}
+      {{- if $.Values.environment }}
+      deployment.environment: {{ $.Values.environment }}
+      {{- end }}
+      {{- if $.Values.extraAttributes.custom }}
+      {{- toYaml $.Values.extraAttributes.custom | nindent 6 }}
+      {{- end }}
```

Comment:
You can use existing `resource` processor, after batching it'll be more performant.

Btw you cannot use `Values.extraAttributes.custom` as is here. it has different format

Reply:
Thanks @dmitryax. I'll update the PR to use existing resource processor
Will still need to keep `host.name` as it's not a part of resource processor.